SC_redesign/Crop_and_Soil/soil/phosphorus_cycling/manure_application.py

```diff
@@ -0,0 +1,168 @@
+        self.data.grazing_stable_inorganic_phosphorus += total_phosphorus_mass * 0.1125
+        self.data.grazing_stable_organic_phosphorus += total_phosphorus_mass * 0.3375
+
+        application_field_coverage = self._determine_grazing_manure_field_coverage(field_size, dry_matter_mass)
+        new_vals = self._determine_weighted_manure_attributes(self.data.grazing_manure_dry_mass,
+                                                              self.data.grazing_manure_moisture_factor,
+                                                              self.data.grazing_manure_field_coverage, dry_matter_mass,
+                                                              dry_matter_content, application_field_coverage)
+        self.data.grazing_manure_dry_mass = new_vals.get("new_dry_matter_mass")
+        self.data.grazing_manure_moisture_factor = new_vals.get("new_moisture_factor")
+        self.data.grazing_manure_field_coverage = new_vals.get("new_field_coverage")
+
+    # --- Static Methods ---
+    @staticmethod
+    def _determine_grazing_manure_field_coverage(field_size: float, total_manure_applied: float) -> float:
+        """Calculates the fraction of the field covered by manure that was applied by grazers.
+
+        Parameters
+        ----------
+        field_size : float
+            Size of the field (ha)
+        total_manure_applied : float
+            Total mass of the manure application (kg)
+
+        Returns
+        -------
+        float
+            The fraction of the field covered by manure (unitless)
+
+        Notes
+        -----
+        This method is only used for calculating the field coverage for manure applied by grazers. It is based on the
+        relationship specified in the SurPhos theoretical documentation which states that the ratio is 250 grams of
+        manure to 659 square centimeters of field coverage (James et al., 2007).
+
+        References
+        ----------
+        James E., Kleinman P., Veith T., Stedman R., Sharpley A. (2007) Phosphorus contributions from
+            pastured dairy cattle to streams of the Cannonsville Watershed, New York. Journal of Soil
+            and Water Conservation 62:40-47.
+
+        """
+        manure_applied_in_grams = total_manure_applied * KILOGRAMS_TO_GRAMS
+        field_coverage_in_square_cm = manure_applied_in_grams * (659 / 250)
+        field_coverage_in_ha = field_coverage_in_square_cm * SQUARE_CENTIMETERS_TO_HECTARES
+        field_coverage_fraction = min(1.0, field_coverage_in_ha / field_size)
+        return field_coverage_fraction
+
+    @staticmethod
+    def _determine_moisture_factor(dry_matter_content: float) -> float:
+        """This method determines the moisture factor of a new manure application based on how much manure was applied
+            and how much water was in the application.
+
+        Parameters
+        ----------
+        dry_matter_content : float
```

Comment:
What about swapping "content" for "fraction"? If you've already been using "content" for fractional variables, then ignore this. Otherwise, it **may** be a bit confusing.

```suggestion
        dry_matter_fraction : float
```

Reply:
Yes, fraction is a lot clearer than content, and content is used to mean something different elsewhere in the program.